pkg/build/api/v1beta1/types.go

```diff
@@ -212,10 +213,11 @@ type CustomBuildStrategy struct {
 	// JSON tag should be changed to "tag" instead of "Tag" in v1beta2
 	Tag string `json:"Tag,omitempty"`
 
-	// PullSecretName is the name of a Secret that would be used for setting up
+	// PullSecret is the name of a Secret that would be used for setting up
 	// the authentication for pulling the Docker images from the private Docker
-	// registries
-	PullSecretName string `json:"pullSecretName,omitempty"`
+	// registries.
+	// This field must reference the resource of kind 'Secret'.
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"supported type: dockercfg"`
```

Comment:
@liggitt I think same as @smarterclayton said before, this stuff was not there long enough that it is worth doing conversion. @smarterclayton I also updated 'v1' API you pushed in recently for this change.